templates/module/rest-api/serializer-template.ts

```diff
@@ -0,0 +1,3 @@
+import { $entityName } from '../types';
+
+export const serialize$entityNameAsJSON = (data: $entityName): unknown => data;
```

Comment:
Can we make this serialize method similar to the one's in boilerplate if possible, ex.
```
export const serializeAccountAsJSON = (account: Account): unknown => ({
  id: account.id,
  phoneNumber: account.phoneNumber,
  firstName: account.firstName,
  lastName: account.lastName,
  username: account.username,
});
```
Because in here we are just changing the type of the object, not exactly serializing it.

Reply:
To serialise it exactly we also need to be aware of the individual property names, but we don't know exactly which field the user wants to send and which he doesn't so here we are just changing the type and letting the user decide which fields the user wants to serialise and send and which don't

Reply:
> To serialise it exactly we also need to be aware of the individual property names

I think this is the same case with defining the type def of the schema too, where we need to know the attribute names. So, we can use similar implementation IMO.

> but we don't know exactly which field the user wants to send and which he doesn't so here we are just changing the type

Usually we were including all the attributes while serialising. In case, where the object's structure matches the desired JSON format exactly and it doesn't contain any sensitive information, we can use the serialise method that you have added. But in contradiction to your statement: `letting the user decide which fields the user wants to serialise and send and which don't` this takes out the control as we cannot choose the fields that we want, and we need to again implement the way that I have stated in the earlier comment. IMO, although it is simple and works well there is also chances that this might
- Exposes the internal structure of the object, which might include unnecessary or sensitive information. 
- Doesn't provide fine-grained control over the serialised output.
- It is also completely ignoring the nested object serialising, ex. let's say there is a resource object inside an account, then this isn't properly serialising the resource object.

So, at the end we might need to implement the template of existing boilerplate's serialiser methods, let's try to align with the same everywhere to keep it consistent. To keep it simple we can just include all the attributes while serialising (I'm fine with this for now, as we can update later with minor efforts), but there are still possibilities of issues 1 and 3. We need to implement some kind of solution like defining the required properties in `yml` file or similar.